torch/csrc/jit/mobile/interpreter.h

```diff
@@ -29,13 +32,17 @@ struct Code {
 };
 
 struct InterpreterState {
-  TORCH_API explicit InterpreterState(std::shared_ptr<Code> code);
+  TORCH_API explicit InterpreterState(const Code& code);
```

Comment:
Any specific reason to change from `shared_ptr` to pass by reference here?

Reply:
InterpreterState doesn't outlive the Function calling it, so it doesn't have to own the Code. 
The general problems of std::shared_ptr are 1. Performance-wise it needs to increment/decrement atomic counters on their own. 2. More importantly interface-wise using them makes us leaking the shared ownership everywhere in codebase, making the future change harder.